test/framework/kubernetesversions/template.go

```diff
@@ -74,6 +74,7 @@ func GenerateCIArtifactsInjectedTemplateForDebian(input GenerateCIArtifactsInjec
 	}
 	input.ArtifactsDirectory = framework.ResolveArtifactsDirectory(input.ArtifactsDirectory)
 	if input.KubeadmConfigTemplateName == "" {
+		// TODO: adapt this to machinepools
```

Comment:
Please open a separate issue for that. We have to think about if we want to do something here but it is absolutely independent of ClusterClass. This function is used to inject a script that installs Kubernetes latest CI binaries and is only used by infra providers

(short issue description is fine. Something like "Consider if we have to adapt GenerateCIArtifactsInjectedTemplateForDebian for MachinePools")

Reply:
I've opened the issue: https://github.com/kubernetes-sigs/cluster-api/issues/9200

This should only leave dropping the comment here :)